pkg/scm/git/git.go

```diff
@@ -62,18 +67,136 @@ func stringInSlice(s string, slice []string) bool {
 // ValidCloneSpec determines if the given string reference points to a valid git
 // repository
 func (h *stiGit) ValidCloneSpec(source string) bool {
-	url, err := url.Parse(source)
-	if err != nil {
+	rc, _, _, _, _, _ := validCloneSpec(source, false, nil)
+	return rc
+}
+
+// ValidCloneSpecRemoteOnly determines if the given string reference points to a valid remote git
+// repository; valid local repository specs will result in a return of false
+func (h *stiGit) ValidCloneSpecRemoteOnly(source string) bool {
+	rc, _, _, _, _, _ := validCloneSpec(source, true, nil)
+	return rc
+}
+
+// MungeNoProtocolURL will take a URL returned from net/url.Parse and make
+// corrections to the URL when no protocol is specified in the Scheme
+func (h *stiGit) MungeNoProtocolURL(source string, uri *url.URL) bool {
+	if uri == nil {
+		return false
+	}
+	rc, scheme, user, host, path, ref := validCloneSpec(source, false, uri)
+	if !rc {
+		return false
+	}
+	if scheme != "ssh" && scheme != "file" {
```

Comment:
this is confusing behavior....  at a minimum i think it needs some comments.

if the uri wasn't a validCloneSpec, the function returns false.  but it also returns false if it was valid, but the protocol was anything other than "file" or "ssh"?  (which is an odd pairing of protocols to be treated the same)